src/Workspaces/Core/Portable/CodeFixesAndRefactorings/IFixAllContext.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using FixAllScope = Microsoft.CodeAnalysis.CodeFixes.FixAllScope;
+
+namespace Microsoft.CodeAnalysis.CodeFixesAndRefactorings
+{
+    internal interface IFixAllContext
+    {
+        IFixAllState State { get; }
+        IFixAllProvider? FixAllProvider { get; }
```

Comment:
def doc this.  how do we have a fix-all-context if we have no fix all provider to perform the operation?

Reply:
Fixed this now to be a non-nullable field. This was coming from public FixAllContext constructors, which do not take a FixAllProvider, while our internal one used by code fix and refactoring engines always create the context with a non-null FixAllProvider.